src/Microsoft.AspNetCore.Rewrite/Operands/Operand.cs

```diff
@@ -0,0 +1,13 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Text.RegularExpressions;
+using Microsoft.Extensions.FileProviders;
+
+namespace Microsoft.AspNetCore.Rewrite.Operands
+{
+    public interface IOperand
+    {
+        bool? CheckOperation(Match previous, string concatTestString, IFileProvider fileProvider);
```

Comment:
3 cases:
the condition matched, the condition didn't match, and there was a syntax error in checking the test (ex parsing an invalid string to an int). I return null such that it will return false from the condition matching, rather than inverting the false into true.